### PR TITLE
added nodejs week1 homework

### PR DESCRIPTION
**Pull request by krithi-kondana**

### CODE REVIEW

File: nodejs/week1/meal_app/src/backend/routes/cheap-meals.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const express = require("express");
+const app = express();
+
+const mealsJson = require(__dirname + '/../data/meals.json');
+const reviewsJson = require(__dirname + '/../data/reviews.json');
+
+const cheapMealsArray = mealsJson.filter(cheapMeals => cheapMeals.price < 50)
```
Comment (Hatko) [inline nodejs/week1/meal_app/src/backend/routes/cheap-meals.js]: A small thing - your naming is misleading here. Inside of `filter` function, you go through `all` meals, so:
1. As meals are not filtered at the moment the function runs, they are not `cheapMeals` yet. They rather all the meals you have.
2. Inside of `filter` function, you are going through every meal, so it's not plural.

I would call it just `meal`.

The same mistake repeats for some routes as well.

File: nodejs/week1/meal_app/src/backend/routes/cheap-meals.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+cheapMealsArray.forEach(meal => {
+    meal.review = [];
```
Comment (Hatko) [inline nodejs/week1/meal_app/src/backend/routes/cheap-meals.js]: You can write it shorter with filter:

```
meal.review = reviewsJson.filter(mealReview => mealReview.mealId === meal.id)
```

If you have questions regarding how it works, ask me on Sunday ;)

The same code applicable to all the places you used similar logic.

File: nodejs/week1/meal_app/src/backend/routes/meal.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    meal.review = [];
+    reviewsJson.forEach(mealreview => {
+        if (mealreview.mealId === meal.id) {
+            meal.review.push(mealreview)
+        }
+    })
+})
+
+app.get("/", (request, response) => {
+    let randomMeal = mealsJson[Math.floor(Math.random() * mealsJson.length)];
```
Comment (Hatko) [inline nodejs/week1/meal_app/src/backend/routes/meal.js]: `let` -> `const`, as it never changes

File: nodejs/week1/meal_app/src/backend/routes/reservation.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const express = require("express");
+const app = express();
+
+const reservationsJson = require(__dirname + '/../data/reservations.json');
+
+app.get("/", (request, response) => {
+    let randomReservation = reservationsJson[Math.floor(Math.random() * reservationsJson.length)];
```
Comment (Hatko) [inline nodejs/week1/meal_app/src/backend/routes/reservation.js]: `let` -> `const`

File: nodejs/week1/meal_app/src/backend/routes/cheap-meals.js (hunk 1 above)
Comment (krithi-kondana) [reply nodejs/week1/meal_app/src/backend/routes/cheap-meals.js]: Thank you Vlad for the feedback :)  I have implemented necessary changes. Waiting for your approval.
regards
krithi

**Pull request merged**